Source/FunicularSwitch.Generators/UnionType/Generator.cs

```diff
@@ -92,7 +92,10 @@ static void WritePartialWithStaticFactories(UnionTypeSchema unionTypeSchema, CSh
         var typeParameters = RoslynExtensions.FormatTypeParameters(unionTypeSchema.TypeParameters);
 
         var typeKind = GetTypeKind(unionTypeSchema);
-        builder.WriteLine($"{(unionTypeSchema.Modifiers.ToSeparatedString(" "))} {typeKind} {unionTypeSchema.TypeName}{typeParameters}");
+        var actualModifiers = unionTypeSchema.Modifiers
+            .Select(m => m == "public" ? unionTypeSchema.IsInternal ? "internal" : "public" : m);
```

Comment:
the double ternary operator is a bit difficult to read, should we wrap the inner one in braces to make it a bit easier to comprehend?
```suggestion
            .Select(m => m == "public" ? (unionTypeSchema.IsInternal ? "internal" : "public") : m);
```